### PR TITLE
per project renamed file handling

### PR DESCRIPTION
**Pull request by vladak**
Introduces per-project tunable to enable override of renamed file handling (this is handy e.g. when doing full reindex when renamed file handling is enabled by default and some projects have too rich history that is not needed. Or vice versa.). After some trial and error I decided to push this information down to `Repository` level so that it is cached.

As a test I ran the indexer with the following read-only config (supplied to the -R option) and without `--renamedHistory` (it will be off by default):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<java version="1.8.0_60" class="java.beans.XMLDecoder">
 <object class="org.opensolaris.opengrok.configuration.Configuration" id="Configuration0">

  <void property="projects">
   <void method="put">
    <string>foo-renamed</string>
    <object class="org.opensolaris.opengrok.configuration.Project">
     <void property="handleRenamedFiles">
      <boolean>true</boolean>
     </void>
    </object>
   </void>
  </void>

 </object>
</java>
```

and checked renamed files for this project have complete history.

Then used the same, reversed the condition to `false`, cloned the project and ran the indexer without `--renamedHistory on`, checked the first project does not have renamed history and the second has.

Also ran the indexer without projects (without the -P option) as a sanity check.

### CODE REVIEW

**Comment by vladak** [comment]
Once this is merged in, I will update https://github.com/oracle/opengrok/wiki/Per-project-configuration

File: src/org/opensolaris/opengrok/configuration/Project.java [hunk 1]
```diff
@@ -68,6 +68,12 @@
      */
     private boolean navigateWindowEnabled = false;
 
+    /**
+     * This flag sets per-project handling of renamed files.
+     */
+    private enum propState { YES, NO, UNSET };
+    private propState handleRenamedFiles = propState.UNSET;
```
Comment (idodeclare) [inline src/org/opensolaris/opengrok/configuration/Project.java]: Hi. Perhaps `Boolean` (i.e. a nullable object wrapping `boolean` and therefore tri-state) would do here just as well?

Comment (vladak) [reply src/org/opensolaris/opengrok/configuration/Project.java]: good idea!

File: src/org/opensolaris/opengrok/configuration/Project.java [hunk 2]
```diff
@@ -229,6 +234,20 @@ public void setNavigateWindowEnabled(boolean navigateWindowEnabled) {
         this.navigateWindowEnabled = navigateWindowEnabled;
     }
 
+    /**
+     * @return true if this project handles renamed files.
+     */
+    public boolean isHandleRenamedFiles() {
+        return handleRenamedFiles;
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/Project.java]: this cause potential NPE when `handleRenamedFiles` is null, doesn't it?

Comment (vladak) [reply src/org/opensolaris/opengrok/configuration/Project.java]: yes, did not mean to expose `null` outside.

**Pull request merged**